website/blog/2022-05-10-acf-in-a-project.md

```diff
@@ -0,0 +1,140 @@
+
+When you define all the fields you need, save them and go to `Custom Fields -> Tools`. Here you will see an option to **_Export Field Groups_**. Simply select the field groups which you want to export and select **_Generate PHP_**.
+
+![ACF PHP code export](/img/blog/acf-generate-php.png)
+
+This will generate a PHP code snippet that you can use in your theme. But now you may be wondering, where exactly should that code go?
+
+## CustomMeta class
+
+Those ACF goodies in Eightshift Development kit I talked about earlier? Let me introduce you to one of them. We have a WP CLI command which we can use to generate a CustomMeta class where we can add our field groups. The command is `wp boilerplate create_acf_meta`. This command has one required parameter, and that is `name`. To create a class that we will use for registering our custom fields, we'll use the following command:
+
+```bash
+wp boilerplate create_acf_meta --name=intro
+```
+
+This command will generate a **_CustomMeta_** folder inside **_src_** folder and add a new file called **_IntroAcfMeta.php_**. Inside that file, you should see the following method:
+```php
+public function fields(): void
+{
+	if (function_exists('acf_add_local_field_group')) {
+		\acf_add_local_field_group([]);
+	}
+}
+```
+
+Go back to your PHP code export for ACF fields and select everything inside the `acf_add_local_field_group()` function and paste it into your function. To make it in line with our coding standards, we have to do the following:
+- replace `array()` with `[]`
+- make every label translatable and escaped - use `esc_html__()`
+- instead of hardcoding the field name, replace it with a class constant
+
+After all these changes, your code should look like this:
+
+```php
+class IntroAcfMeta extends AbstractAcfMeta
+{
+
+	/**
+	 * Intro field name.
+	 */
+	public const INTRO_FIELD_NAME = 'intro';
+	
+	/**
+	 * Render acf fields.
+	 *
+	 * @return void
+	 */
+	public function fields(): void
+	{
+		if (function_exists('acf_add_local_field_group')) {
+			\acf_add_local_field_group([
+				'key' => 'group_6269300acefda',
+				'title' =>  \esc_html__('Intro', 'eightshift-theme'),
+				'fields' => [
+					[
+						'key' => 'field_6269300f8029b',
+						'label' => \esc_html__('Intro', 'eightshift-theme'),
+						'name' => self::INTRO_FIELD_NAME,
+						'type' => 'text',
+						// ...
+					]
+				]
+				// ...
+			]);
+		}
+	}
+}
+```
+
+The final step is to go back to the Custom Fields in WP Admin and either delete or deactivate your field group from there, to prevent registering the fields twice. After adding field definitions in PHP and removing them in WP Admin, your field should be visible when editing Posts.
```

Comment:
```suggestion
The final step is to go back to the Custom Fields in WP admin and either delete or deactivate your field group from there, to prevent registering the fields twice. After adding field definitions in PHP and removing them in WP admin, your field should be visible when editing posts.
```